client/src/controllers/KeyboardController.ts

```diff
@@ -0,0 +1,32 @@
+import { Controller } from '@hotwired/stimulus';
+import Mousetrap from 'mousetrap';
+
+/**
+ * Adds the ability to trigger a button click event using a keyboard shortcut
+ * declared on the controlled element.
+ *
+ * @example
+ * ```html
+ * <button type="button" data-controller="w-kbd" data-w-kbd-key="[">Trigger me with the <kbd>[</kbd> key.</button>
+ * ```
+ */
+export class KeyboardController extends Controller<HTMLButtonElement> {
+  static values = { key: { default: '', type: String } };
+
+  handleKey(event: Event) {
+    if (event.preventDefault) event.preventDefault();
+    this.element.click();
+  }
+
+  initialize() {
+    this.handleKey = this.handleKey.bind(this);
```

Comment:
If you are up for it, I would like to suggest we add basic support for using https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-keyshortcuts

```suggestion
    this.handleKey = this.handleKey.bind(this);
    if (!this.keyValue && this.element.ariaKeyShortcuts) {
        this.keyValue = this.element.ariaKeyShortcuts;
    }
```

This would need a unit test also but basically when the controller initalizes we check if there is a set `keyValue`, if not, we set the `keyValue` to the ariaKeyShortcuts.

This means, for basic shortcuts, we could do this.

```html
<button type="button" aria-keyshortcuts="[" data-controller="w-kbd">Open side menu</button>
```

Once the JS loads, this will automatically set the key-value and behave as if that's what was set originally.

```html
<button type="button" aria-keyshortcuts="[" data-controller="w-kbd" data-w-kbd-key-value="[">Open side menu</button>
```

This would not work for the 'mod' support as this is not an officially supported modifier, it's just a MouseTrap abstraction on (Command on MacOS, Control on Windows). But it would work for other simple shortcuts and reinforces accessibility.

One day, in the future, we could work the other way with setting the the `aria-keyshortcuts` value on the element if it's not set but the controller is in use. However, again, we would have to add more complex logic to support this to account for some shortcut strings that Mousetrap uses that are not in the HTML spec.

It's good to note that this may not be fully supported by all browsers yet though https://a11ysupport.io/tech/aria/aria-keyshortcuts_attribute

Anyway... I think a small change in the `initialize` method and another unit test is worth it for a nice improvement... if you are keen :D

Reply:
I'll look into it 🙌 
I think I should also add a test for `mod+j` because I couldn't get it to trigger when I send a `KeyboardEvent` with `ctrl:True` and `keyCode:j`